core/src/main/java/com/sequenceiq/cloudbreak/job/StackStatusCheckerJob.java

```diff
@@ -246,6 +246,7 @@ private void ifFlowNotRunning(Runnable function) {
         return EnumSet.of(
```

Comment:
To make sure I understand what's happening, `updateStates` checks this enum for a set of values that should automatically be switched to `AVAILABLE`, rather than whatever end state the cluster is in.

If so, this looks good to me.

Reply:
Yes, exactly what you said. The sync job sync every 3 minutes to bring the states in that ENUMS to available. We put the backup_failed in it so it won't block further datalake operation.